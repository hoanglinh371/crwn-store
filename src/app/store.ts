import { configureStore, ThunkAction, Action } from '@reduxjs/toolkit';

import cartReducer from '../features/cart/redux/cart.slice';
import categoryReducer from '../features/category/redux/category.slice';
import productReducer from '../features/product/redux/product.slice';

export const store = configureStore({
  reducer: {
    productReducer,
    cartReducer,
    categoryReducer,
  },
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
