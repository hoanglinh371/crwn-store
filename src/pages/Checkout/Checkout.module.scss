.container {
  width: 55%;
  min-height: 90vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  margin: 50px auto 0;

  @media screen and (max-width: 600px) {
    width: 100%;
  }

  .header {
    width: 100%;
    padding: 10px 0;
    display: flex;
    justify-content: space-between;
    border-bottom: 1px solid darkgrey;

    .header-block {
      text-transform: capitalize;
    }
  }

  .total {
    margin-top: 30px;
    margin-left: auto;
    font-size: 2.25rem;
  }
}
