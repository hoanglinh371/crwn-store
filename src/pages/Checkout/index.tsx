import React from 'react';
import classNames from 'classnames/bind';

import styles from './Checkout.module.scss';

import CheckoutItem from 'src/components/CheckoutItem';

const cx = classNames.bind(styles);

const Checkout: React.FC = () => {
    const cartItems: any = [];

    return (
        <div className={cx('container')}>
            <div className={cx('header')}>
                <div className={cx('header-block')}>
                    <span>Product</span>
                </div>
                <div className={cx('header-block')}>
                    <span>Description</span>
                </div>
                <div className={cx('header-block')}>
                    <span>Quantity</span>
                </div>
                <div className={cx('header-block')}>
                    <span>Price</span>
                </div>
                <div className={cx('header-block')}>
                    <span>Remove</span>
                </div>
            </div>
            {cartItems.map((item: any) => (
                <CheckoutItem key={item.id} cartItem={item} />
            ))}
        </div>
    );
};

export default Checkout;
