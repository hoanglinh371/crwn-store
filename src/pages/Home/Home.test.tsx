import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import { cleanup, render } from '@testing-library/react';

import { store } from 'src/app/store';

import Checkout from '.';

describe('Home component', () => {
  afterEach(cleanup);

  it('Should_MatchSnapshot', () => {
    const { container } = render(
      <Provider store={store}>
        <BrowserRouter>
          <Checkout />
        </BrowserRouter>
      </Provider>,
    );
    expect(container).toMatchSnapshot();
  });
});
