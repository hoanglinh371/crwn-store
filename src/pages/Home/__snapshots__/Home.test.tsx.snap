// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component Should_MatchSnapshot 1`] = `
<div>
  <div
    class="container"
  />
</div>
`;
